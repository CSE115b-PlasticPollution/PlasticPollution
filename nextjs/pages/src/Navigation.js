import logo from './assets/ucsc.png';
import React from 'react';
import Nav from 'react-bootstrap/Nav';
import { NavLink } from 'react-router-dom';
import Navbar from 'react-bootstrap/Navbar';
import NavDropdown from 'react-bootstrap/NavDropdown';
import './Navigation.css';
import Head from 'next/head'

class Navigation extends React.Component{
	render() {
		return(
			<div>
					<Navbar bg="light" expand="lg" collapseOnSelect>
					  <Navbar.Brand href="/"><img src={logo} className="logo"></img></Navbar.Brand>
			  		<Navbar.Toggle aria-controls="responsive-navbar-nav"/>
					   <Navbar.Collapse id="responsive-navbar-nav">
					    <Nav className="mr-auto">
					      <Nav.Link href="/">Home</Nav.Link>
					      <Nav.Link href="/about">About</Nav.Link>
								<Nav.Link href="/faq">FAQ</Nav.Link>
								<Nav.Link href="/Contact">Contact</Nav.Link>
								<Nav.Link href="/Activist">Become an Activist</Nav.Link>
								<NavDropdown title="Start an Adventure" id="collasible-nav-dropdown">
								<NavDropdown.Item href="/Macro">Macro</NavDropdown.Item>
								<NavDropdown.Item href="/Micro">Micro</NavDropdown.Item>
								<NavDropdown.Item href="/Nano">Nano</NavDropdown.Item>
							  </NavDropdown>
					    </Nav>
					  </Navbar.Collapse>
					</Navbar>
      </div>
				)
		}
}

export default Navigation;
