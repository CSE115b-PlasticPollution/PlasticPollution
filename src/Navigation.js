import logo from './assets/ucsc.png';
import React from 'react';
import Nav from 'react-bootstrap/Nav';
import { NavLink } from 'react-router-dom';
import Navbar from 'react-bootstrap/Navbar';
import NavDropdown from 'react-bootstrap/NavDropdown';
import './Navigation.css';
import {Link } from "react-router-dom";

class Navigation extends React.Component{
	render() {
		return(
			<div>
					<Navbar bg="light" expand="lg" collapseOnSelect>
					  <Navbar.Brand href="/"><img src={logo} className="logo"></img></Navbar.Brand>
			  		<Navbar.Toggle aria-controls="responsive-navbar-nav"/>
					   <Navbar.Collapse id="responsive-navbar-nav">
					    <Nav className="mr-auto">
					      <Nav.Link href="/https://cse115b-plasticpollution.github.io/plasticpollution/#/">Home</Nav.Link>
					      <Nav.Link href="/plasticpollution/#/about">About</Nav.Link>
								<Nav.Link href="/https://cse115b-plasticpollution.github.io/plasticpollution/#/faq">FAQ</Nav.Link>
								<Nav.Link href="/https://cse115b-plasticpollution.github.io/plasticpollution/#/Contact">Contact</Nav.Link>
								<Nav.Link href="/https://cse115b-plasticpollution.github.io/plasticpollution/#/Activist">Become an Activist</Nav.Link>
								<NavDropdown title="Start an Adventure" id="collasible-nav-dropdown">
								<NavDropdown.Item href="/https://cse115b-plasticpollution.github.io/plasticpollution/#/Macro">Macro</NavDropdown.Item>
								<NavDropdown.Item href="/https://cse115b-plasticpollution.github.io/plasticpollution/#/Micro">Micro</NavDropdown.Item>
								<NavDropdown.Item href="/plasticpollution/#/Nano">Nano</NavDropdown.Item>
							  </NavDropdown>
					    </Nav>
					  </Navbar.Collapse>
					</Navbar>
      </div>
				)
		}
}

export default Navigation;
