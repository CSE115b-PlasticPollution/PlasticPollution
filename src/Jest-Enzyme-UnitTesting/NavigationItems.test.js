import React from 'react';
import renderer from 'react-test-renderer';
import { NavLink } from 'react-router-dom';
import Nav from 'react-bootstrap/Nav';
/* End of npm packages */
/* Jest & Enzyme dependcies */
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
/* Routing duplicate components into Unit-testing foloder to test individually */
import NavigationItems from './NavigationItems.test.js'
import Navigation from './Navigation'

configure({ adapter: new Adapter() });
/* Checks length of object, must be greater then 0*/
it('Plastic Pollution: <NavigationItems /> must render length of object if not authenticated!', () => {
    const wrapper = shallow(<Navigation />);
    expect(wrapper.find(Navigation)).toHaveLength(Navigation.length);

});
/* If we rendered an object to screen, return true*/
describe('<NavigationItems />', () => {
    it('Plastic Pollution: <NavigationItems /> must render elements if not authenticated!', () => {
        const wrapper = shallow(<Navigation />);
        expect(wrapper.find(Navigation)).toBeTruthy();
   
    });
/* Checks if javascript object matches a subset of properties of an object*/
    it('Plastic Pollution: <NavigationItems /> must render object if not authenticated!', () => {
        const wrapper = shallow(<Navigation />);
        expect(wrapper.find(Navigation)).toMatchObject(NavigationItems);
       
    });
/* Checking if links are navigating correctly through github */
    it('Github Live Links render correctly', () => {
        const tree = renderer
            .create(<Nav.Link href= "/https://cse115b-plasticpollution.github.io/plasticpollution/#/">Landing Page</Nav.Link>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    /* Checking if links are navigating correctly locally */
    it('Local Links render correctly', () => {
        const tree = renderer
            .create(<Nav.Link href="/plasticpollution/#/">Landing Page</Nav.Link>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});