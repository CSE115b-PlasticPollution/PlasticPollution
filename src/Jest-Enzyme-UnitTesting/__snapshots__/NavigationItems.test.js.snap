// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems /> Github Live Links render correctly 1`] = `
<a
  className="nav-link"
  href="/https://cse115b-plasticpollution.github.io/plasticpollution/#/"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  Landing Page
</a>
`;

exports[`<NavigationItems /> Local Links render correctly 1`] = `
<a
  className="nav-link"
  href="/plasticpollution/#/"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  Landing Page
</a>
`;
